compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -1190,8 +1207,14 @@ impl<'a> Parser<'a> {
             }
         }
 
-        let poly_trait =
-            PolyTraitRef::new(lifetime_defs, path, modifiers, lo.to(self.prev_token.span));
+        let grouping = if has_parens { ast::Parens::Yes } else { ast::Parens::No };
```

Comment:
Would be nice if callers to this also used `ast::Parens` instead of passing a bool. If you want to do it that's fine by me, but I can do it as a followup.